sdcm/utils/time_utils.py

```diff
@@ -0,0 +1,27 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2024 ScyllaDB
+
+from datetime import datetime, timedelta
+
+
+class ExecutionTimer:
+    def __enter__(self):
+        self.start_time = datetime.now()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.end_time = datetime.now()
+
+    @property
+    def duration(self) -> timedelta:
+        return self.end_time - self.start_time
```

Comment:
IIRC we have something else already that you can use here.

Reply:
I tried to find something like this but didn't

Reply:
I think something like "duration".
If you want to expand it and measure also latency, you should have latency_calculator which measures both latency and duration.